fiftyone/core/stages.py

```diff
@@ -951,8 +952,8 @@ def _get_meta_filtered_fields(sample_collection, meta_filter, frames=False):
         )
     )
 
-    type_matcher = (
-        lambda query, field: (
+    type_matcher = lambda query, field: (
```

Comment:
Same, I find this lambda incredibly confusing. I recommend refactoring to a named function instead

```python
  def type_matcher(query, field):
    if isinstance(query, str):
      if isinstance(field, EmbeddedDocumentField):
        return type(field.document_type).__name__ == query or field.document_type.__name__ == query
      else:
        return type(field).__name__ == query
    elif isinstance(field, EmbeddedDocumentField):
      return isinstance(field.document_type, query) or field.document_type == query
    else:
      return isinstance(field, query)
```

Reply:
The diff here and the one above are only due to formatting. I can update, but it has no affect on the focus of the PR

Reply:
I know. Your choice on whether you want to refactor in this PR or not. Just calling it out.